### PR TITLE
Avoiding cuda compiler error in debug

### PR DESCRIPTION
**Pull request by rrsettgast**

### CODE REVIEW

File: src/indexing.hpp [hunk 1]
```diff
@@ -186,12 +186,15 @@ LVARRAY_HOST_DEVICE inline constexpr
...
+  int * const p_curDim = &curDim;
+
   bool invalid = false;
-  typeManipulation::forEachArg( [dims, &curDim, &invalid]( auto const index )
-  {
-    invalid = invalid || ( index < 0 ) || ( index >= dims[ curDim ] );
-    ++curDim;
-  }, indices ... );
+  bool * const p_invalid = &invalid;
+  typeManipulation::forEachArg( [dims, p_curDim, p_invalid] LVARRAY_HOST_DEVICE ( INDEX_TYPE const index )
```
Comment (corbett5) [inline src/indexing.hpp:193]: I guess you can't capture by reference in host device lambdas?

**Review by corbett5 — APPROVED** [review]
Gotta love how this just now shows up two years later...

File: src/indexing.hpp (hunk 1 above)
Comment (rrsettgast) [reply src/indexing.hpp:193]: I guess not. It makes sense as it would cause a race condition?

Comment (corbett5) [reply src/indexing.hpp:193]: No race condition here. Each CUDA thread creates its own copy of the lambda. Just another silly NVCC restriction that only occasionally shows up.

Comment (rrsettgast) [reply src/indexing.hpp:193]: *could cause a race condition

**Pull request closed without merging**